libs/langchain/tests/unit_tests/chat_models/test_huggingface_llama2.py

```diff
@@ -0,0 +1,37 @@
+"""Test Hugging Face Llama-2 Chat model."""
+
+from langchain.chat_models.huggingface_llama2 import ChatLlama2Hf
+from langchain.schema.messages import (
+    AIMessage,
+    HumanMessage,
+    SystemMessage,
+)
+
+
+def test_format_messages_as_text_with_system() -> None:
+    messages = [
+        SystemMessage(content="System Prompt."),
+        HumanMessage(content="Human Message."),
+        AIMessage(content="AI response."),
+        HumanMessage(content="Second Human Message."),
+        AIMessage(content="AI response."),
+    ]
+
+    ground_truth = "<s>[INST] <<SYS>>\nSystem Prompt.\n<</SYS>>\n\nHuman Message. [/INST] AI response. </s><s>[INST] Second Human Message. [/INST] AI response. </s><s>[INST] "  # noqa: E501
+
+    messages_as_str = ChatLlama2Hf.format_messages_as_text(messages=messages)
+    assert messages_as_str == ground_truth
+
+
+def test_format_messages_as_text_without_system() -> None:
+    messages = [
+        HumanMessage(content="Human Message."),
+        AIMessage(content="AI response."),
+        HumanMessage(content="Second Human Message."),
+        AIMessage(content="Second AI response."),
+    ]
+
+    ground_truth = "<s>[INST] Human Message. [/INST] AI response. </s><s>[INST] Second Human Message. [/INST] Second AI response. </s><s>[INST] "  # noqa: E501
```

Comment:
Possible it will break if defined by you tokens were to be changed. Probably not likely but still you might want to assert that those special tokens are indeed equal to [INST] etc - maybe add test for it, to at least allow diagnose of such issue easy by failing on exact problem.